sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py

```diff
@@ -0,0 +1,451 @@
+        print "    {FINGER_NAME}J{JOINT_NUMBER} - for testing individual joints (e.g. FFJ3, RFJ4 etc...)"
+        print "  OR:"
+        print "    J{JOINT_NUMBER}              - for testing the same joint on all fingers (e.g. J3, J4)"
+        print ""
+        print "                                 - If COMMAND_CHAR is 'p', using J{JOINT_NUMBER} will"
+        print "                                   move all joints in sequence to min, then max, then zero"
+        print ""
+        print "                                 - If COMMAND_CHAR is 'e', using J{JOINT_NUMBER} will"
+        print "                                   move apply PWM for three seconds, then -PWM for three seconds"
+        print ""
+        print "  {ANGLE/PWM/LIMIT} can be:"
+        print "    when COMMAND_CHAR is 'p' - this value is taken as an angle (if a number is specified)"
+        print "                             - or as a command to move to joint limit (e.g. 'min', 'max')"
+        print ""
+        print "    when COMMAND_CHAR is 'e' - This value is taken as a direct PWM command"
+        print ""
+        print "Examples:"
+        print "  e_ffj3_200 - Apply a constant PWM of 200 to joint FFJ3"
+        print "  e_ffj3_-240 - Apply a constant PWM of -240 to joint FFJ3"
+        print "  p_ffj3_60  - Move ffj3 to 60 degrees"
+        print "  p_ffj3_-5  - Move ffj3 to -5 degrees"
+        print "  p_j3_0     - Test all joint 3's  (position mode)  in fingers: FF, RF, MF, LF"
+        print "  e_j3_150   - Test all joint 3's (direct PWM mode) in fingers: FF, RF, MF, LF"
+        print "  p_ffj4_min - Move FFJ4 to it's minimum angle"
+        print "  p_rfj3_max - Move RFJ3 to it's maximum angle"
+        print ""
+        print "(Note: Using 'p' or 'e' in COMMAND_CHAR will change ALL controllers to "
+        print "position or effort, not just the controllers for the finger specified)"
+        print ""
+
+    def write_output_dictionary(self, filename_prefix, filename, output_dictionary, dictionary_keys):
+        fieldnames = []
+        rows = []
+        output_filename = filename_prefix + filename
+        for key in dictionary_keys:
+            fieldnames.append(key)
+        length_of_list = len(output_dictionary[dictionary_keys[0]])
+        for i in range(0, length_of_list):
+            line = {}
+            for key in dictionary_keys:
+                line[key] = output_dictionary[key][i]
+            rows.append(line)
+        self.write_csv(output_filename, fieldnames, rows)
+
+    def write_csv(self, filename, fieldnames, rows):
+        with open(filename, mode='w') as csv_file:
+            writer = csv.DictWriter(csv_file, fieldnames=fieldnames)
+            writer.writeheader()
+            for row in rows:
+                writer.writerow(row)
+        rospy.loginfo("file: %s saved", filename)
+
+
+if __name__ == '__main__':
+    rospy.init_node("sr_test_force_resolution")
+    test_force_resolution = TestForceResolution()
```

Comment:
we should get the params locally. and launch this via a launch file or at least be able to pass the arguments via terminal.

Reply:
e.g. side = left can only be set by modifying it here.